### PR TITLE
[Test-Proxy] Support Extensible Storage

### LINKED ISSUE #3324: Add Extensibility Options for `Storage` Retrieval
Currently, the test-proxy functions purely in its role as a proxy. A "context" is GIVEN to it. However, with this transition to external assets, we now have a couple additional entries in the devs loop:

- Install Local Test Requirements
- Install Test Proxy
- **Get External Assets Retrieval Tool or Script**
- **Call External Assets Retrieval** 
- Run Tests

Given that we are _already_ forcing devs to install the test proxy as either a docker image or nuget package, we would truly like to avoid giving them _yet another_ external tool to install. By adding these capabilities as a purely optional feature of the test-proxy, we can centralize the responsibility and avoid adding another dependency to the install loop.

## Enter Storage Extensions

We need to expose the following routes via API:

- `/Playback/Load`  - to pull from an external storage location (we will start with git repo external storage)
- `/Record/Save` - to save to the external storage location

The users should pass the target assets.json file in the request body. `AssetsJson: <absolute-path-on-disk>`.

We ideally would like to provide this same capability exposed as a CLI command:

- `test-proxy save <path-to-assets-json>`
- `test-proxy load <path-to-assets-json>`

Which should utilize the same code as the live routes.  These routes should call via interface, with the "default" storage implementation being a null-op. This will give us the power to either change this default storage implementation via `Admin/SetRecordingOptions/` or by entry in the input `assets.json`.

### PR DESCRIPTION
**Pull request by scbedd**
Fixes #3324

- [x] Add command line interface for access to functionality from new API Endpoints
- [x] Add Interface layer that will enable optional data stores, implement and default to basic `Do nothing` until `GitStore` is written 
- [x] Implement "resolve store class instance from assembly"
- [x] ...and activate" via `SetRecordingOptions` key of `"AssetsStore"` to allow users to change which store is being used
- [x] Add CLI option to select target store at startup
- [x] Add Restore/Save/Reset API Endpoints
~~- [ ] Document storage functionality in readme~~ I'll document it when adding the _actual_ functionality that will work. All that it has is `NoOp` operations right now.
- [x] Tests for new API Layer

### CODE REVIEW

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/IAssetsStore.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+namespace Azure.Sdk.Tools.TestProxy.Store
+{
+    public interface IAssetsStore
+    {
+        /// <summary>
+        /// Given a configuration, push the changes made by the test-proxy into the remote store.
+        /// </summary>
+        /// <param name="pathToAssetsJson"></param>
+        /// <param name="contextPath"></param>
+        public abstract void Save(string pathToAssetsJson, string contextPath);
```
Comment (benbp) [inline tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/IAssetsStore.cs:12]: Is `remote store` in this case just whatever local git/disk provider is being used, or does it mean "push to a shared upstream" remote?

Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/IAssetsStore.cs:12]: The latter; pushed to a shared upstream remote. This equivalent to an `Push` operation in the asset-sync script.

Comment (benbp) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/IAssetsStore.cs:12]: In this case I'd be a little worried that `Save` does not communicate that effectively? Or is this just an internal name/route?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Startup.cs [hunk 2]
```diff
@@ -136,7 +162,14 @@ public void ConfigureServices(IServiceCollection services)
             });
             services.AddControllersWithViews();
             services.AddRazorPages();
-            services.AddSingleton<RecordingHandler>(new RecordingHandler(TargetLocation));
+
+            var singleTonRecordingHandler = new RecordingHandler(
```
[the remark this answers is not in the document]
Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Startup.cs:166]: This may be a bug, I gotta check it :P

**Review by benbp — APPROVED** [review]

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/IAssetsStore.cs (hunk 1 above)
Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Store/IAssetsStore.cs:12]: What do you think will be more appropriate here?

I am thinking of `save` as "place in external storage" whether it's git or not. Make it `push`?

File: tools/test-proxy/Azure.Sdk.Tools.TestProxy/Startup.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (scbedd) [reply tools/test-proxy/Azure.Sdk.Tools.TestProxy/Startup.cs:166]: No. Definitely not lol. I see what you're calling out now. Fixing.

**Pull request merged**